### ISSUE TITLE
Error spans.

### ISSUE THREAD

**Issue by thedodd** [jtescher/tracing-opentelemetry#17]
First, I would just like to highlight the excellent work done here! I've been watching progress on this ecosystem for a decent bit of time. You've made some awesome progress! Thanks @jtescher!

So, Jaeger will only show spans as being error spans if it holds the tag `error=true`. I am wondering if perhaps there is something which we should be doing here, or perhaps in the opentelemtry-jaeger crate (over in the main repo) to ensure that spans are being shown properly when they are error spans.

As a technicality, I almost never instrument my code with an actual error span, as in `tracing::error_span!`. Typically it will be a `tracing::error!` event, which is perhaps part of the issue (I'm not sure, I haven't jumped into the code yet). 

Either way, I wanted to pop an issue here to start the discussion. Def let me know what you think. Cheers!

**Comment by jtescher** [comment]
@thedodd yes good points, I've been meaning to add this! Looks like the way it would work is if you log an attribute that impls `std::error::Error`, then the expected behavior would be to have jaeger report the span as errored. It should be pretty straightforward to implement this as well, are you interested in submitting a pr?

Looks like one solution would be to implement the `record_error` method from https://github.com/tokio-rs/tracing/blob/351ac21809d929c6cdf4d7134ce7e9a70df212eb/tracing-core/src/field.rs#L208-L210. 

Something like changing `SpanEventVisitor` to:

```rust
struct SpanEventVisitor<'a> {
    event: &'a mut api::Event,
    builder: &'a mut api::SpanBuilder,
}
```

And adding the status code to the builder during `record_error`:

```rust
if self.builder.status_code.is_none() {
    self.builder.status_code = Some(api::StatusCode::Unknown);
}
if self.builder.status_message.is_none() {
    self.builder.status_message = Some(value.to_string());
}
```

The jaeger exporter will add `error=true` for any span that has a no-`OK` status code.

**Comment by jtescher** [comment]
Resolved in #18, we can reopen if additional methods of signaling failure state are needed

**Comment by thedodd** [comment]
@jtescher you rock! Thank you for knocking that out so quickly.
